shared/src/commonMain/kotlin/org/hyperskill/app/gamification_toolbar/presentation/GamificationToolbarActionDispatcher.kt

```diff
@@ -78,15 +113,59 @@ class GamificationToolbarActionDispatcher(
                 onNewMessage(
                     Message.FetchGamificationToolbarDataSuccess(
                         streak,
-                        profile.gamification.hypercoinsBalance
+                        profile.gamification.hypercoinsBalance,
+                        trackWithProgress
                     )
                 )
             }
+            is Action.FetchTrackWithProgress -> {
+                sentryInteractor.startTransaction(action.transaction)
+                fetchTrackWithProgress(action.trackId, true)
+                    .fold(
+                        onSuccess = { trackWithProgress ->
+                            onNewMessage(Message.FetchTrackWithProgressResult.Success(trackWithProgress))
+                            sentryInteractor.finishTransaction(action.transaction)
+                        },
+                        onFailure = {
+                            onNewMessage(Message.FetchTrackWithProgressResult.Error)
+                            sentryInteractor.finishTransaction(action.transaction)
+                        }
+                    )
+            }
             is Action.LogAnalyticEvent ->
                 analyticInteractor.logEvent(action.analyticEvent)
             else -> {
                 // no op
             }
         }
     }
+
+    private suspend fun fetchTrackWithProgressThoughtStudyPlan(forceLoadFromRemote: Boolean): Result<TrackWithProgress?> =
+        kotlin.runCatching {
+            val studyPlan =
+                currentStudyPlanStateRepository.getState(forceLoadFromRemote).getOrThrow()
+            if (studyPlan.trackId != null) {
+                fetchTrackWithProgress(studyPlan.trackId, forceLoadFromRemote).getOrThrow()
+            } else {
+                null
+            }
+        }
+
+    private suspend fun fetchTrackWithProgress(trackId: Long, forceLoadFromRemote: Boolean): Result<TrackWithProgress?> =
+        coroutineScope {
+            kotlin.runCatching {
+                val trackDeferred = async {
+                    trackRepository.getTrack(trackId, forceLoadFromRemote)
+                }
+                val trackProgressDeferred = async {
+                    progressesRepository
+                        .getTrackProgress(trackId, forceLoadFromRemote)
+                }
+                TrackWithProgress(
+                    track = trackDeferred.await().getOrThrow(),
+                    trackProgress = trackProgressDeferred.await().getOrThrow()
+                        ?: return@runCatching null
```

Comment:
```suggestion
                    trackProgress = trackProgressDeferred.await().getOrThrow() ?: return@runCatching null
```

Reply:
It's more readable when in one line

<img width="702" alt="Screenshot 2023-04-10 at 2 06 01 PM" src="https://user-images.githubusercontent.com/8586063/230890263-f6f5de0d-a86a-4f88-8c7e-2d24deec2828.png">